alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -150,10 +149,22 @@ bool BasicBehaviour::isFailure() const
 
 void BasicBehaviour::setTrigger(essentials::ITrigger* trigger)
 {
+    if (!trigger)
+        return;
+
+    std::lock_guard<std::mutex> lockGuard(_runLoopMutex);
     _behaviourTrigger = trigger;
     _behaviourTrigger->registerCV(&_runCV);
 }
 
+bool BasicBehaviour::isTriggeredRunFinished()
+{
+    if (!_behaviourTrigger)
```

Comment:
_behaviourTrigger is guarded in setTrigger, but not here? Is there something that is ensuring that the guard is not required here?

Reply:
added lock_guard